lib/improver/utilities/spatial.py

```diff
@@ -56,47 +56,62 @@ def check_if_grid_is_equal_area(cube):
         cube (iris.cube.Cube):
             Cube with coordinates that will be cMAhecked.
     Raises:
-        ValueError : Invalid grid: projection_x/y coords required
-        ValueError : Intervals between points along the x and y axis vary.
-                     Therefore the grid is not an equal area grid.
-        ValueError : The size of the intervals along the x and y axis
-                     should be equal.
+        ValueError: If coordinate points are not equally spaced along either
+            axis (from calculate_grid_spacing)
+        ValueError: If point spacing is not equal for the two spatial axes
     """
+    xdiff = calculate_grid_spacing(cube, axis='x')
+    ydiff = calculate_grid_spacing(cube, axis='y')
+    if not np.isclose(xdiff, ydiff):
+        raise ValueError("Grid is not equal area")
```

Comment:
I've updated the error message.  To make this function more "honest", I could put in a default argument `require_equal_spacing=True`, so that it is possible to switch off this error if we literally just want an equal area grid.